modules/core/src/geojson-types.js

```diff
@@ -59,7 +65,7 @@ export type BoundingBoxArray = [number, number, number, number];
 export type FeatureOf<T: Geometry> = {
   type: 'Feature',
   geometry: T,
-  properties?: {},
+  properties?: { index?: number },
```

Comment:
@supersonicclay I addressed this issue in my latest diff. I also added test cases to ensure that properties provided by nebula users are preserved. I also renamed index to _internalIndex to avoid key collisions.